nextjs-app/app/services/likesAndDislikesService/likesAndDislikes.ts

```
@@ -0,0 +1,97 @@
+import { getDbInstance } from "@/db/utils";
+import { closeDatabase } from "@/db/initDatabase";
+import { NextResponse } from "next/server";
+
+enum VoteType {
+    LIKE = "like",
+    DISLIKE = "dislike",
+    LIKED = "liked",
+    DISLIKED = "disliked"
+}
+
+interface PlayerVote {
+    userSteamId: string;
+    otherPlayerSteamId: string;
+    type: VoteType;
+}
+
+export async function putLikesAndDislikes({ userSteamId, otherPlayerSteamId, type }: PlayerVote) {
+    const db = await getDbInstance();
+    try {
+        
+        await new Promise((resolve, reject) => db.run("BEGIN TRANSACTION", (err) => (err ? reject(err) : resolve(true))));
+        
+        const existingVote: Array<Record<string, any>> = await new Promise((resolve, reject) => {
+            db.all(
+                `SELECT likes_dislikes FROM likeDislike WHERE steam_id = ? AND other_player_steam_id = ?`,
+                [String(userSteamId), String(otherPlayerSteamId)],
+                (err, rows) => {
+                    if (err) {
+                        console.error("Error executing query:", err);
+                        return reject(err);
+                    }
+                    resolve(rows as any);
+                }
+            );
+        });
+
+        if (type === VoteType.LIKED || type === VoteType.DISLIKED) {
+            // If type is liked or disliked, set likes_dislikes to NULL
+            await new Promise((resolve, reject) => {
```

should we check if existing vote is correct? 
If we are trying to liked(im assuming to remove like) then LIKE vote should be present, if not that is invalid case and we should throw an error

same logic for dislike and removal of dislike

---

Added